MudDataGrid: Accessibility fixes

OllieFox131 opened the pull request:
Adding aria-labels to fix accessibility issues resulting in a significant reduction in wave tool errors on the MudDataGrid component page

With Changes:

![image](https://github.com/user-attachments/assets/607b47ba-8d21-4e63-892f-bb327196bacb)

Without:

![image](https://github.com/user-attachments/assets/78cd2743-aac8-41db-853a-ad0f2c8ab6cc)

Aria Labels have been added to
- Filter Header Cell
  - Clear Filter
- Header Cell
  - Sort Changed
  - Open Filters
  - Add Filter
  - Show Column Options
- Mud Data Grid
  - Toggle Group Expansion
  - Remove Filter
- Mud Data Grid Pager
  - First Page
  - Previous Page
  - Next Page
  - Last Page   

## Description
Following the rejection of https://github.com/MudBlazor/MudBlazor/pull/5719

The fixes would have been useful to fix many of the empty button error messages on the data grid

I have put together some changes that hopefully reintroduces those fixes but in a style more similar to that which gets accepted in https://github.com/MudBlazor/MudBlazor/pull/9064

## How Has This Been Tested?
visually and WAVE tool 

## Type of Changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation (fix or improvement to the website or code docs)

## Checklist
- [x] The PR is submitted to the correct branch (`dev`).
- [x] My code follows the code style of this project.
- [ ] I've added relevant tests.

Commit:
add aria-labels

Commit:
add more aria-labels

Commit:
Fixes

Commit:
Fix spacing

Commit:
Misssing Space

ScarletKuro commented:
cc @danielchalmers @igotinfected to check accessibility
cc @meenzen for new translation keys

danielchalmers commented:
The icon buttons are good, but regular buttons that already have text should not have an aria label

```razor
<MudButton Variant="Variant.Filled" Color="Color.Default" OnClick="@CancelEditingItemAsync" Class="px-10" aria-label="@Localizer[LanguageResource.MudDataGrid_CancelEditingItem]">@Localizer[LanguageResource.MudDataGrid_Cancel]</MudButton>
```

See https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-label

Commit:
Remove non-icon buttons

Commit:
Dont change urls!

OllieFox131 commented:
> The icon buttons are good, but regular buttons that already have text should not have an aria label
> 
> ```
> <MudButton Variant="Variant.Filled" Color="Color.Default" OnClick="@CancelEditingItemAsync" Class="px-10" aria-label="@Localizer[LanguageResource.MudDataGrid_CancelEditingItem]">@Localizer[LanguageResource.MudDataGrid_Cancel]</MudButton>
> ```
> 
> See https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-label

Updated and removed new aria-labels from MudButtons

meenzen reviewed (approved):
Translation handling looks good

henon commented:
Shall we let our accessibility expert @igotinfected take a look before merging?

ScarletKuro commented:
> Shall we let our accessibility expert @igotinfected take a look before merging?

Yes. But we need to get patience as:
> Bit busy with life right now

igotinfected commented:
LGTM!
